api/hooks.ts

```diff
@@ -1,25 +1,237 @@
 import { APIConfig, FastifyTypebox } from '.'
-import Store from '../store'
-
-export const hookRoutes = (cfg: APIConfig, store: Store) => async (server: FastifyTypebox): Promise<void> => {
-  // Register hooks for new inbox items that get added to the moderation queue
-  // Hooks are an array of {url, method, headers}
-  // The body will contain the inbox item
-  server.get('/:domain/hooks/onmoderationqueued', async (request, reply) => { })
-  server.delete('/:domain/hooks/onmoderationqueued', async (request, reply) => { })
-  server.put('/:domain/hooks/onmoderationqueued', async (request, reply) => { })
-
-  // Register hooks for new inbox items that get approved
-  // Hooks are an array of {url, method, headers}
-  // The body will contain the inbox item
-  server.get('/:domain/hooks/onnew', async (request, reply) => { })
-  server.delete('/:domain/hooks/onnew', async (request, reply) => { })
-  server.put('/:domain/hooks/onnew', async (request, reply) => { })
-
-  // Register hooks for new inbox items that get rejected
-  // Hooks are an array of {url, method, headers}
-  // The body will contain the inbox item
-  server.get('/:domain/hooks/onrejected', async (request, reply) => { })
-  server.delete('/:domain/hooks/onrejected', async (request, reply) => { })
-  server.put('/:domain/hooks/onrejected', async (request, reply) => { })
+import { Type } from '@sinclair/typebox'
+import { HookStore } from '../store/HookStore'
+
+export const hookRoutes = (cfg: APIConfig, store: HookStore) => async (server: FastifyTypebox): Promise<void> => {
+  // For ModerationQueued
+  server.put('/:actor/hooks/moderationqueued', {
+    schema: {
+      params: Type.Object({
+        actor: Type.String()
+      }),
+      body: Type.Object({
+        url: Type.String({
+          format: 'uri'
+        }),
+        method: Type.Union([Type.Literal('GET'), Type.Literal('POST'), Type.Literal('PUT'), Type.Literal('DELETE')]),
+        headers: Type.Record(Type.String(), Type.String())
+      }),
+      response: {
+        200: Type.Object({
+          message: Type.String()
+        })
+      },
+      description: 'Sets a hook for when an item is added to the moderation queue.',
+      tags: ['Hooks']
+    }
+  }, async (request, reply) => {
+    // const domain = request.params.domain;
+    const hook = request.body
+    await store.setModerationQueued(hook)
```

Comment:
The hook store should be set on the actor store.
```suggestion
    await store.forActor(actor).setModerationQueued(hook)
```